client/src/components/Ticketing/ticketingmanager/TicketTypes/tickettypes.tsx

```diff
@@ -1,60 +1,216 @@
+          newRow.price !== prevRow.price) {
+          console.log('Ticket type has changed');
+
+          const response = await fetch(
+              process.env.REACT_APP_ROOT_URL+'/api/tickets/updateType', {
+                method: 'PUT',
+                headers: {
+                  'Content-Type': 'application/json',
+                  'Authorization': `Bearer ${token}`,
+                },
+                body: JSON.stringify(newRow),
+              },
+          );
+          console.log(response);
+        }
+        return newRow;
+      }, [ticketTypes],
+  );
 
   // handles the click event of the delete button
-  const handleDeleteClick = () => {
+  const handleDeleteClick = async (cell: GridRenderCellParams) => {
     console.log('Delete Clicked');
+    const ticketId = Number(cell.row.id);
+    ticketId.toString();
+
+    const token = await getAccessTokenSilently({
+      audience: 'https://localhost:8000',
+      scope: 'admin',
+    });
+
+    try {
+      const response = await fetch(
+          process.env.REACT_APP_ROOT_URL + '/api/tickets/' + ticketId, {
+            method: 'DELETE',
+            headers: {
+              'Content-Type': 'application/json',
+              'Authorization': `Bearer ${token}`,
+            },
+          });
+
+      const responseData = await response.json();
+      console.log(responseData);
+      getTicketTypes(); // refreshes the page ie re-renders the table
+    } catch (error) {
+      console.log(error);
+    }
   };
 
+  const handleNameChange = (event) => {
+    setTicketType(event.target.value);
+  };
+
+  const handlePriceChange = (event) => {
+    setTicketPrice(event.target.value);
+  };
+
+  const handleConcessionsChange = (event) => {
+    setConcessionsPrice(event.target.value);
+  };
+
+  const handleSubmit = async () => {
+    console.log('Submit clicked');
+
+    const token = await getAccessTokenSilently({
+      audience: 'https://localhost:8000',
+      scope: 'admin',
+    });
+
+    try {
+      const response = await fetch(
+          process.env.REACT_APP_ROOT_URL + '/api/tickets/newType',
+          {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+              'Authorization': `Bearer ${token}`,
+            },
+            body: JSON.stringify(
+                {
+                  name: newTicketType,
+                  price: newTicketPrice,
+                  concessions: newConcessionsPrice,
+                },
+            ),
+          },
+      );
+      if (response.ok) {
+        console.log('Ticket type added successfully');
+        setAddTicketClicked(!addTicketClicked); // closes the form
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  // Shows the form to add a new ticket type
+  const showAddTicketView = () => {
+    return (
+      <div className='bg-blue-200 rounded-xl p-10 shadow-md mb-4'>
+        <div className='shadow-xl p-5 rounded-xl mb-9 bg-blue-700'>
```

Comment:
We should use `bg-violet-200` and `bg-violet-700` respectively, to be consistent with the manage events page (found in `showingInputContainer`)